specs/online_order_spec.rb

```diff
@@ -2,70 +2,133 @@
 require 'minitest/reporters'
 require 'minitest/skip_dsl'
 
-# TODO: uncomment the next line once you start wave 3
-# require_relative '../lib/online_order'
+require_relative '../lib/online_order'
+require_relative '../lib/order'
+
+require 'pry'
 # You may also need to require other classes here
 
 # Because an OnlineOrder is a kind of Order, and we've
 # already tested a bunch of functionality on Order,
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
+  before do
+    @online_order = Grocery::OnlineOrder.new(1000, {apples: 2.00, bananas: 1.00}, 30)
+  end
+
   describe "#initialize" do
-    it "Is a kind of Order" do
-      # Check that an OnlineOrder is in fact a kind of Order
 
+    it "Is a kind of Order" do
       # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      @online_order.must_be_kind_of Grocery::Order
     end
 
     it "Can access Customer object" do
-      # TODO: Your test code here!
+
+      @online_order.must_respond_to :customer_id
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      @online_order.must_respond_to :status
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      sum = 0
+      products = @online_order.products
+
+      products.each do |key, value|
+          sum += value
+      end
+      sum_incl_tax = (sum + (sum * 0.075)).round(2)
+      @online_order.total.must_equal (sum_incl_tax + 10).round(2)
     end
 
+###TODO
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      new_online_order = Grocery::OnlineOrder.new(1000, 0, 30)
+      new_online_order.total.must_equal 0
     end
   end
 
   describe "#add_product" do
+    before do
+      @new_online_orders = []
+      @new_online_orders << Grocery::OnlineOrder.new(1000, {apples: 2.00, bananas: 1.00}, 30, :completed)
+      @new_online_orders << Grocery::OnlineOrder.new(1000, {apples: 2.00, bananas: 1.00}, 30, :shipped)
```

Comment:
While I like the idea, because the set of orders needed for each test is not the same I'm not sure that a `before` block is the right tool here. It might be cleaner to redefine these orders in each of the tests.